graph_pattern.py

```diff
@@ -709,6 +711,90 @@ def to_count_var_over_values_query(self, var, vars_, values, limit):
         res += 'LIMIT %d\n' % limit
         return self._sparql_prefix(res)
 
+    def to_find_edge_var_for_narrow_path_query(self, edge_var, node_var,
+                                               vars_, filter_node_count,
+                                               filter_edge_count, limit_res):
+        """Counts possible substitutions for edge_var to get a narrow path
+
+        Meant to perform a query like this:
+        SELECT *
+        {
+          {
+            SELECT
+              ?edge_var
+              (COUNT(*) AS ?edge_var_count)
+              (MAX(?node_var_count) AS ?max_node_count)
+              (COUNT(*)/AVG(?node_var_count) as ?prio_var)
+            {
+              SELECT DISTINCT
+                ?source ?target ?edge_var (COUNT(?node_var) AS ?node_var_count)
+              {
+                VALUES (?source ?target) {
+                  (dbr:Adolescence dbr:Youth)
+                  (dbr:Adult dbr:Child)
+                  (dbr:Angel dbr:Heaven)
+                  (dbr:Arithmetic dbr:Mathematics)
+                }
+                ?node_var ?edge_var ?source .
+                ?source dbo:wikiPageWikiLink ?target .
+              }
+            }
+            GROUP BY ?edge_var
+            ORDER BY DESC(?edge_var_count)
+          }
+          FILTER(?max_node_count < 10 && ?edge_var_count > 1)
+        }
+        ORDER BY DESC(?prio_var)
+        LIMIT 32
+
+        :param edge_var: Edge variable to find substitution for.
+        :param node_var: Node variable to count.
+        :param vars_: List of vars to fix values for (e.g. ?source, ?target).
+        :param values: List of value lists for vars_.
+        :param filter_node_count: Filter on node count of edge variable.
+        :param filter_edge_count: Filter for edge count of triples.
+        :param limit_res : limit result size
+        :return: Query String.
+        """
+
+        res = 'SELECT * WHERE {\n'
+        res += ' {\n'\
+               '  SELECT %s (COUNT(*) as %s) (Max(%s) AS %s) ' \
+               '   (COUNT(*)/AVG(%s) AS %s) WHERE {\n' % (
+                     edge_var.n3(), EDGE_VAR_COUNT.n3(),
+                     NODE_VAR_COUNT.n3(), MAX_NODE_COUNT.n3(),
+                     NODE_VAR_COUNT.n3(), PRIO_VAR.n3())
+        res += '    SELECT DISTINCT %s %s (COUNT(%s) AS %s) WHERE {\n' % (
+                   ' '.join([v.n3() for v in vars_]),
+                   edge_var.n3(), node_var.n3(), NODE_VAR_COUNT.n3())
+        # res += self._sparql_values_part(values)
+        res += 'VALUES(%s) {\n' \
+               '(dbr: Adolescence dbr: Youth)' \
+               '(dbr:Adult dbr:Child)' \
+               '(dbr:Angel  dbr:Heaven)' \
+               '(dbr:Arithmetic dbr:Mathematics)' \
+               '}\n' % (' '.join([v.n3() for v in vars_]))
```

Comment:
that should be replaced by the `values` part... also `dbr:<SPACE>Anything` won't work...